tests/unit/C/requirements.sh

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+
+##--------------------------------------------------------------------
+## Copyright (c) 2025 Dianomic Systems
+##
+## Licensed under the Apache License, Version 2.0 (the "License");
+## you may not use this file except in compliance with the License.
+## You may obtain a copy of the License at
+##
+##     http://www.apache.org/licenses/LICENSE-2.0
+##
+## Unless required by applicable law or agreed to in writing, software
+## distributed under the License is distributed on an "AS IS" BASIS,
+## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+##--------------------------------------------------------------------
+
+##
+## Author: Ashwini Kumar Pandey
+##
+
+set -e  # Exit immediately if a command exits with a non-zero status.
+
+# Function to install and build Google Test for Ubuntu
+install_gtest_ubuntu_from_package() {
+    echo "Installing Google Test for Ubuntu..."
+
+    # Install the libgtest-dev package
+    sudo apt-get update -y
+    sudo apt-get install -y libgtest-dev
+
+   if [[ ${OS_VERSION} == *"18."* ]]; then
+        echo "Building Google Test libraries manually..."
+        cd /usr/src/gtest
+        sudo cmake -E make_directory build
+        sudo cmake -E chdir build cmake ..
+        sudo cmake --build build
+        sudo cp build/libgtest* /usr/lib
+    fi
+
+    echo "Google Test installation complete for Ubuntu..."
+}
+
+# Function to build and install Google Test from source
+install_gtest_ubuntu_from_source() {
+    echo "Installing Google Test for using source..."
+
+    # Define repository name and branch
+    local GTEST_REPO_NAME="googletest"
+    local GTEST_BRANCH="${1:-release-1.11.0}"  # Default branch is release-1.11.0
+
+    # Clean up any existing directory
+    if [ -d "/tmp/${GTEST_REPO_NAME}" ]; then
+        echo "Removing existing ${GTEST_REPO_NAME} directory in /tmp..."
+        sudo rm -rf "/tmp/${GTEST_REPO_NAME}"
+    fi
+
+    # Clone the Google Test repository
+    echo "Cloning Google Test repository (${GTEST_BRANCH})..."
+    git clone https://github.com/google/${GTEST_REPO_NAME}.git --branch="${GTEST_BRANCH}" --depth 1 -q /tmp/${GTEST_REPO_NAME}
+
+    # Build and install Google Test
+    cd "/tmp/${GTEST_REPO_NAME}"
+    mkdir -p build && cd build
+    echo "Configuring and building Google Test..."
+    cmake .. -DBUILD_GMOCK=OFF > /dev/null
+    make -j$(nproc) > /dev/null
+
+    echo "Installing Google Test..."
+    sudo make install > /dev/null
+
+    echo "Google Test installation complete for Ubuntu 24.04 or later!"
+}
+
+# Function to install Google Test for Red Hat-based distributions
+install_gtest_rhel() {
+    echo "Installing Google Test for Red Hat-based distributions..."
+
+    # Install required packages
+    sudo yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
```

Comment:
This information is significantly outdated, as it was established for RHEL/CentOS 7 support. We no longer offer support for this version and are now concentrating on CentOS Stream 9 and future releases, although no specific plans have been made yet. Therefore, I suggest removing this line to prevent any confusion when we revisit Red Hat support.